apps/chat/channels.py

```diff
@@ -185,28 +187,23 @@ def _chat_initiated(self):
         """The user initiated the chat and we need to get their consent before continuing the conversation"""
         self.experiment_session.update_status(SessionStatus.PENDING_PRE_SURVEY)
         consent_text = self.experiment.consent_form.consent_text
-        conversational_consent = self.experiment.consent_form.conversational_consent
-        conversational_consent = conversational_consent.replace("{", "")
-        conversational_consent = conversational_consent.replace("}", "")
-        consent_text = f"{consent_text}\n{conversational_consent}"  # TODO: Find a better way
-        return self._send_message_as_bot(consent_text)
+        confirmation_text = self.experiment.consent_form.confirmation_text
+        return self._send_message_as_bot(f"{consent_text}\n\n{confirmation_text}")
 
     def _send_message_as_bot(self, message: str):
         """Send a message to the user as the bot and adds it to the chat history"""
         self._add_message_to_history(message, ChatMessageType.AI)
         self.send_text_to_user(message)
 
-    def _should_get_consent(self):
+    def _should_prompt_for_consent(self):
         """Pre-conversation is the phase where the user starts the chat and gives consent to continue"""
-        return self.experiment_session.status in [SessionStatus.SETUP, SessionStatus.PENDING_PRE_SURVEY]
+        return self.experiment.conversational_consent_enabled and self.experiment_session.status in [
+            SessionStatus.SETUP,
+            SessionStatus.PENDING_PRE_SURVEY,
+        ]
 
     def _user_gave_consent(self) -> bool:
-        """Match the user's input to the configured "acceptance" keywords"""
-        consent_text = self.experiment.consent_form.conversational_consent
-        pattern = r"\{(.*?)\}"
-        matches = re.findall(pattern, consent_text)
-        consent_keywords = matches[0].split(",")
-        return self.message_text in [word.lstrip() for word in consent_keywords]
+        return self.message_text == USER_CONSENT_TEXT
```

Comment:
should probably do `self.message_text.strip()`

Reply:
https://github.com/dimagi/open-chat-studio/pull/151/commits/e2af2e254ee356789eac5782538076d6709185fd